modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -357,6 +357,8 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
         virtual ~Layer();
     };
 
+    struct LayerData;
```

Comment:
This is an internal data structure.

Reply:
https://github.com/opencv/opencv/pull/16015#discussion_r351744027
To create a seprate functon from the common parts in the two loading method (file and buffer):
- I have to pass an InferenceEngine::CNNNetReader which can not be included in the dnn.hpp, so i had to use static function
- I need to access the Impl struct of the Net class which is forward declarated in dnn.hpp and this where i can't find a clean solution.

As I see now I have these options here: 
1. Writing a getter to member impl which doesn't seems right because its a private member. 
2. The current solution: writing a getter which can get a layerData from the impl struct by id.
To do this a had to forward declarate the LayerData struct in the header. 
3. Writing setters to layerInstance and backendNodes:
        ld.layerInstance = cvLayer;
        ld.backendNodes[DNN_BACKEND_INFERENCE_ENGINE] = backendNode;
The problem here is the backendNode which is has an InfEngineBackendNet type whisch also can't be included in the dnn.hpp 

If there is an other way please help me to find it.